lib/sync/pool.go

```diff
@@ -0,0 +1,75 @@
+package sync
+
+import (
+	"context"
+	"errors"
+	"sync"
+)
+
+type Pool struct {
+	finish chan struct{}
+	work   chan<- func()
+	wg     sync.WaitGroup
+}
+
+var ErrFinished = errors.New("Add: Pool is finished")
+
+func NewPool(n int) *Pool {
+	work := make(chan func())
+	finish := make(chan struct{})
+	p := &Pool{
+		work:   work,
+		finish: finish,
+	}
+
+	for ; n > 0; n-- {
+		p.wg.Add(1)
```

Comment:
Please use channels instead of a WaitGroup of 1

Reply:
@Geod24 Why?

Reply:
It's more consistent with the rest of sebak, it usually looks better, and it gets rid of one dependency.
The [documentation on sync](https://golang.org/pkg/sync/) itself mentions that channel are higher level.

Reply:
I am not sure I understand what you mean @anarcher ... But it's not such a big point, so if you feel strongly about it, let's leave it as is!

Reply:
@anarcher : "Let's leave it as is" == Ok, no changes ;)

> Saying higher level thing is a meaningless point.

Higher level usually means more streamlined design (easier to read/reason about), which is where I was getting at here.

Reply:
> Higher level usually means more streamlined design (easier to read/reason about), which is where I was getting at here.

@Geod24 It depends on. and Not always higher level makes easier to read/reason about. And sync pkg is higher level at some point (synchronized way) than channel/goroutine.
I can make waitgroup with channels/goroutines without `sync/lock and mutex and atomic.` Opposite too ... Which is higher?